stream.go

```diff
@@ -236,7 +236,7 @@ type StreamReader interface {
 	//
 	// If no new entries are available or there was an error, ok will be false.
```

Comment:
Sorry I didn't see this earlier (and that this review has had so many revisions >_<). I think ok should only be false if there's an error. The user can see that there's no new entries by checking if the returned slice is empty. This docs string should probably mention explicitly the empty slice case so users know they need to handle it, but I think it makes the logic a bit clearer for the user to do:

```go
for {
    stream, entries, ok := streamReader.Next()
    if !ok {
        return streamReader.Err()
    } else if len(entries) == 0 {
        continue
    }
    // ...
}
```

rather than:

```go
// the current way would look like this
for {
    stream, entries, ok := streamReader.Next()
    if !ok {
        if err := streamReader.Err(); err != nil {
            return err
        }
        continue
    }
    // ...
}
```

Reply:
Done